src/value.cc

```diff
@@ -315,18 +274,35 @@ bool Value::getFiniteDouble(double &v) const
   return valid;
 }
 
+Value::VectorPtr::VectorPtr(double x, double y, double z) : ptr(make_shared<Value::VectorType>()) {
+  (*this)->emplace_back(x);
```

Comment:
Uncertain about this, but since the `vector` constructor takes an initializer list, could you just to `: ptr(make_shared<Value::VectorType>(x, y, z)) {`

Reply:
I think it's possible but much too complicated to be worth it. The problem is that the value type of `VectorType` is `Value` which is move-only. That does not easily work with initializer lists. There are some example that make it work with `make_moveable_iterator` but that seems  too messy for this case.